### DISCUSSION TITLE
Outdated self-hosted docker images

### DISCUSSION THREAD

**Discussion by derevyan** [supabase/supabase#27608] [Questions]
I'm experimenting with a self-hosted Supabase using the default configuration instructions from the Docker folder on the main GitHub branch. I discovered that most of the Docker images are outdated and point to older specific versions. Typically, production-ready apps point to the :latest tag, allowing for easy updates with docker pull. This makes me hesitant to use Supabase in production. Does anyone have experience maintaining an up-to-date Supabase? How do you manage it?

For example - Supabase uses the following image:
`image: kong:2.8.1`
This version of the Kong package is from **two years ago**, whereas the current version is **3.6.1.**

Thanks.

**Comment by C5G6M7** [comment]
Hi,

I agree Supabase should be packaged with a more recent version of Kong especially considering how simple it is to make the adjustment. Also as of Kong 3.4 it includes the Kong Manager OSS GUI which is a nice feature to have. There are additional Kong features you can enable as well such as DB mode which allows dynamically changing configurations instead of having to declare them all up front in the kong.yml file. I was able to get this working by editing the supabase-kong container in the docker-compose.yml file. 

One thing I discovered is that on the newer versions of kong the file permissions are different inside the container that prohibit write access to the home folder so it is necessary to point the kong config files to /usr/local/kong/ as that folder is owned by the kong user. In my version below you can see that I also activated the ports for the Kong Manager GUI and provided the DB details.

If you want to use DB mode just make sure to run the following command inside the container after starting it for the first time:

`kong config db_import /usr/local/kong/kong.yml`

This will load the kong configuration into the database.

Here is the updated supabase-kong container inside the docker-compose.yml file (please excuse the bad formatting):

  kong:
    container_name: supabase-kong
    image: "${KONG_DOCKER_TAG:-kong:latest}"
    restart: unless-stopped
    # https://unix.stackexchange.com/a/294837
    entrypoint: bash -c 'eval "echo \"$$(cat /usr/local/kong/kong-temp.yml)\"" > usr/local/kong/kong.yml && /docker-entrypoint.sh kong docker-start'
    healthcheck:
      test: [ "CMD", "kong", "health" ]
      interval: 10s
      timeout: 10s
      retries: 10
    networks:
      - kong-net
    ports:
      - ${KONG_HTTP_PORT}:8000/tcp
      - ${KONG_HTTPS_PORT}:8443/tcp
      - ${KONG_ADMIN_API_PORT}:8001
      - ${KONG_ADMIN_API_SECURE_PORT}:8444
      - ${KONG_ADMIN_GUI_PORT}:8002
      - ${KONG_ADMIN_GUI_SECURE_PORT}:8445
    depends_on:
      analytics:
        condition: service_healthy
    environment:
      KONG_DECLARATIVE_CONFIG: /usr/local/kong/kong.yml
      # https://github.com/supabase/cli/issues/14
      KONG_DATABASE: ${KONG_DATABASE:-off}
      KONG_PG_HOST: ${POSTGRES_HOST:-db}
      KONG_PG_PORT: ${POSTGRES_PORT}
      KONG_PG_USER: ${POSTGRES_USER}
      KONG_PG_PASSWORD: ${POSTGRES_PASSWORD}
      KONG_PG_DATABASE: ${POSTGRES_DB}
      KONG_PG_SCHEMA: ${KONG_PG_SCHEMA:-kong}
      KONG_DNS_ORDER: LAST,A,CNAME
      KONG_PLUGINS: request-transformer,cors,key-auth,acl,basic-auth,session,oauth2,jwt,acme,rate-limiting,request-size-limiting,prometheus,pre-function,post-function,request-termination,correlation-id,request-transformer,response-transformer,grpc-gateway,grpc-web
      KONG_NGINX_PROXY_PROXY_BUFFER_SIZE: 160k
      KONG_NGINX_PROXY_PROXY_BUFFERS: 64 160k
      SUPABASE_ANON_KEY: ${ANON_KEY}
      SUPABASE_SERVICE_KEY: ${SERVICE_ROLE_KEY}
      DASHBOARD_USERNAME: ${DASHBOARD_USERNAME}
      DASHBOARD_PASSWORD: ${DASHBOARD_PASSWORD}
      KONG_ADMIN_LISTEN: "0.0.0.0:8001, 0.0.0.0:8444 ssl"
      KONG_ADMIN_GUI_LISTEN: "0.0.0.0:8002, 0.0.0.0:8445 ssl"
      KONG_ADMIN_GUI_URL: http://localhost:8002
    volumes:
      # https://github.com/supabase/supabase/issues/12661
      - ./volumes/api/kong.yml:/usr/local/kong/kong-temp.yml


You will also need to add the following to your .env file which configures the container to use the Kong version 3.7.1-ubuntu - just replace this with whichever images it most relevant to your OS. Also note that I set the schema to "gateway" but you could set this to "kong" or "public" depending on your preferences. Then when you log into the supabase studio you will be able to select the schema from the table viewer and see all of the data in there related to Kong.

KONG_DATABASE=postgres
KONG_PG_SCHEMA=gateway
KONG_DOCKER_TAG=kong:3.7.1-ubuntu
KONG_HTTP_PORT=8000
KONG_HTTPS_PORT=8443
KONG_ADMIN_API_PORT=8001
KONG_ADMIN_API_SECURE_PORT=8444
KONG_ADMIN_GUI_PORT=8002
KONG_ADMIN_GUI_SECURE_PORT=8445

**edited to fix typos
